farcaster-frame/contracts/evm/solidity/CanvasGame.sol

```diff
@@ -33,34 +39,36 @@ contract CanvasGame is Ownable {
     }
 
     function paint(uint256 canvas, uint24 color) public payable {
-        require(msg.value >= fee, "Sufficient fee required to paint");
-        require(canvas < canvasOwner.length, "Canvas ID must exist");
-        require(paintCount[canvas] < paintLimit, "Canvas must not be full");
+        if (!(msg.value >= fee)) revert InsufficientFee();//msg.value, fee); // Sufficient fee required to paint
+        if (!(canvas < canvasOwner.length)) revert CanvasDoesNotExist();//canvas, canvasOwner.length); // Canvas ID must exist
+        if (!(paintCount[canvas] < paintLimit)) revert CanvasFull();//canvas, paintLimit); // Canvas must not be full
```

Comment:
Please flip the conditions to not use negation unnecessarily - `x < y` instead of `!(x >= y)` etc.